### ISSUE TITLE
State of KMP-Testbench

### ISSUE THREAD

**Issue by DrewCarlson** [DrewCarlson/KMP-Testbench#19]
Testbench is very early in development and shouldn't be used to replace any debugging tools that currently work for your projects.  That said it can still be useful for creating project specific debugging tools or supplementing existing tools to better debug unsupported platforms (native targets in particular).

There are a number of important things missing from the tool:
- Secure pairing and communication
- Documentation is incomplete and API docs are non-existent
- Broad and consistent desktop UI components
- Various basic plugins are incomplete or not implemented

Many things are still undecided and likely to change:
- I'm interested in exploring [kotlinx-rpc](https://github.com/Kotlin/kotlinx-rpc) for server/client plugin communication
- [Jewel](https://github.com/jetBrains/jewel) is nice to provide visual consistency across development tools, but it's not fully baked and depends on non-standard maven repos. It may be better to implement a custom theme or use material.
- A basic data-table API is available, but it's implementation needs a lot of work which may change the API shape.
- The gradle plugins are more MVP than complete plugins, they may change significantly when trying to make them production ready.

In summary:  Don't expect a release anytime soon, but feel free to explore snapshot builds and provide feedback.  If you have thoughts, issues, or want to assist any of the above work, contributions and feedback are welcome.
